lib/Recur/RRuleIterator.php

```diff
@@ -588,6 +588,11 @@ protected function nextYearly() {
                     (int)$currentDayOfMonth
                 );
 
+                // maximum year 10000 should be safe to stop (2016 + max 3500 yearly occurrences)
+                if ((int)$currentYear > 10000) {
```

Comment:
We have a setting for this, that might be a bit better:

https://github.com/fruux/sabre-vobject/blob/master/lib/Settings.php#L37